syntax = "proto3";

package wood.mike.se;

service SecretEscapes {
  // unary
  rpc GetSeSale (SeSaleRequest) returns (SeSaleReply);

  // server streaming
  rpc GetAllSeSales(SeSaleRangeRequest) returns (stream SeSaleReply);
}

message SeSaleRequest {
  string id = 1;
}

message SeSaleRangeRequest {
  string start = 1;
  string end = 2;
}

message SeSaleReply {
  string id = 1;
  string url_slug = 2;
  string start = 3;
  string end = 4;
}